library/src/scala/tasty/reflect/SymbolOps.scala

```diff
@@ -82,6 +82,9 @@ trait SymbolOps extends Core {
     def isAnonymousFunction(implicit ctx: Context): Boolean = kernel.Symbol_isAnonymousFunction(self)
     def isAbstractType(implicit ctx: Context): Boolean = kernel.Symbol_isAbstractType(self)
     def isClassConstructor(implicit ctx: Context): Boolean = kernel.Symbol_isClassConstructor(self)
+
+    def isType(implicit ctx: Context): Boolean = kernel.Symbol_isType(self)
+    def isTerm(implicit ctx: Context): Boolean = kernel.Symbol_isTerm(self)
```

Comment:
Here I'm not sure which name to give, should I change it to `isTypeOrClassSymbol`, given that in Tasty Reflect they are different concepts? But then, it looks not so nice compared to `isTerm`.

Reply:
Maybe we should rename the current concepts first to allow us to use these more intuitive names. I will work on that.

Reply:
I created #6129 to change the names

Reply:
The second step will be to convert
```
 *  +- Symbol --+- PackageDefSymbol
 *              +- ClassDefSymbol
 *              +- TypeDefSymbol
 *              +- DefDefSymbol
 *              +- ValDefSymbol
 *              +- BindDefSymbol
 *              +- NoSymbol
```

into

```
 *  +- Symbol --+- PackageDefSymbol
 *              |
 *              +- TypeSymbol -+- ClassDefSymbol
 *              |              +- TypeDefSymbol
 *              |              +- TypeBindDefSymbol
 *              |                           
 *              +- TermSymbol -+- DefDefSymbol
 *              |              +- ValDefSymbol
 *              |              +- BindDefSymbol
 *              +- NoSymbol
```

Reply:
Which I added in #6131

Reply:
Now you should rebase on master